python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -40,14 +51,58 @@
 UPSCALING_INITIAL_NUM_NODES = 5
 
 
+class UtilizationScore(Protocol):
```

Comment:
it would be quite hard. currently the default returns `Tuple[float, float]`. and i don't practically know how to map that to a single float easily :p

Reply:
ah yeah, right

Maybe describe the default example then, to justify the abstraction